modules/dataframe/src/io/csvreader.cpp

```diff
@@ -608,11 +615,18 @@ std::shared_ptr<DataFrame> CSVReader::readData(std::istream& stream) const {
 
     if (firstRowHeader_) {
         rows.erase(rows.begin());
-        // remove lines skipped by filters
-        rows.erase(
-            std::remove_if(rows.begin(), rows.end(),
-                           [&](const auto& row) { return skipRow(row.first, row.second, false); }),
-            rows.end());
+
+        // demangle duplicate column headers similar to Pandas' default behavior
+        // @see https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html
+        std::unordered_map<std::string, int> headerCount;
+        for (auto& header : headers) {
+            if (int count = ++headerCount[header]; count > 1) {
+                header.append(fmt::format(".{}", count - 1));
+            }
+        }
+        if (headerCount.size() != headers.size()) {
+            LogWarn("Duplicate column headers detected. Affected headers are demangled.");
```

Comment:
I would rather either document that we always do this. when we parse the header, or we make it an option. Logging is pointless.

Reply:
generally I think one should be extremely careful with changing any data that the user enters without the user explicitly asking for that change. And unless we are trying to enforce unique column headers I see no point. And if we try to do that, that should mostly likely be done in a different PR after some discussion...

Reply:
We could add an option "Require unique headers" to make it more explicit

Reply:
unless we enforce that a data frame has unique header names, I don't see the point

Reply:
Removed duplicate header handling for now.